src/index.ts

```diff
@@ -1,61 +1,148 @@
-import { Reshuffle, BaseConnector, EventConfiguration } from 'reshuffle-base-connector'
+import {
+  BaseHttpConnector,
+  EventConfiguration,
+  Reshuffle,
+} from '@reshuffle/base-connector'
+import {
+  parseFilter,
+  unwrapDates,
+  wrapDates,
+} from "./util"
 
-export interface _CONNECTOR_NAME_ConnectorConfigOptions {
-  var1: string
-  // ...
+import {Request, Response} from 'express'
+
+const DEFAULT_WEBHOOK_PATH = '/webhooks/wix'
+
+export interface WixConnectorConfigOptions {
+  secret?: string,
+  webhookPath: string,
+}
+
+export type WixAction =
+  'provision'
+  | 'schemas/find'
+  | 'schemas/list'
+  | 'data/get'
+  | 'data/count'
+  | 'data/find'
+  | 'data/insert'
+  | 'data/update'
+
+export interface WixConnectorEventOptions {
+  action: WixAction
 }
 
-export interface _CONNECTOR_NAME_ConnectorEventOptions {
-  option1?: string
-  // ...
+export interface WixRequestContext {
+  settings: Record<string, any>
+  instanceId: string
+  installationId: string
+  memberId: string
+  role: string
 }
 
-export default class _CONNECTOR_NAME_Connector extends BaseConnector<
-  _CONNECTOR_NAME_ConnectorConfigOptions,
-  _CONNECTOR_NAME_ConnectorEventOptions
-> {
-  // Your class variables
-  var1: string
+export interface WixEvent {
+  requestContext: WixRequestContext
+  collectionName?: string
+  filter?: string
+  sort?: any
+  skip?: number
+  limit?: number
+  itemId?: string
+  item?: any
+  body?: any
+  action: WixAction
+  request: Request
+  response: Response
+}
+
+class WixConnector extends BaseHttpConnector<WixConnectorConfigOptions,
+  WixConnectorEventOptions> {
+
+  private readonly secret: string | undefined
+  private webhookUrl: string = ''
 
-  constructor(app: Reshuffle, options?: _CONNECTOR_NAME_ConnectorConfigOptions, id?: string) {
+  constructor(app: Reshuffle, options?: WixConnectorConfigOptions, id?: string) {
     super(app, options, id)
-    this.var1 = options?.var1 || 'initial value'
-    // ...
+    this.secret = options?.secret || undefined
   }
 
   onStart(): void {
-    // If you need to do something specific on start, otherwise remove this function
-  }
-
-  onStop(): void {
-    // If you need to do something specific on stop, otherwise remove this function
+    if (Object.keys(this.eventConfigurations).length) {
+      this.webhookUrl = (this.configOptions.webhookPath || DEFAULT_WEBHOOK_PATH)
```

Comment:
cont webhookUrl

Reply:
hmm techically it is a webhookPath, not Url, the base isn't part of it

Reply:
Changed.